### PR TITLE
[keymgr] Advance keymgr to d2

### PR DESCRIPTION
**Pull request by tjaychen**
Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM with 2 questions
1. the sideload interface to aes and hmac hasn't been finalized, right?
2. can you fix this keymgr related issue #4899?

File: hw/ip/keymgr/data/keymgr.prj.hjson [hunk 1]
```diff
@@ -13,6 +13,12 @@
         life_stage:         "L1",
         design_stage:       "D1",
         verification_stage: "V1",
```
Comment (imphil) [inline hw/ip/keymgr/data/keymgr.prj.hjson:15]: If you increment the version number, can you add a commit hash for the previous version (or not increment the version number).

File: hw/ip/keymgr/doc/checklist.md [hunk 2]
```diff
@@ -40,25 +40,25 @@ Code Quality  | [LINT_SETUP][]                 | Done        |
...
-Security      | [SEC_CM_IMPLEMENTED][]  | Not Started |
-Security      | [SEC_NON_RESET_FLOPS][] | Not Started |
-Security      | [SEC_SHADOW_REGS][]     | Not Started |
-Security      | [SEC_RND_CNST][]        | Not Started |
+Documentation | [NEW_FEATURES][]        | Done        |
+Documentation | [BLOCK_DIAGRAM][]       | Done        |
+Documentation | [DOC_INTERFACE][]       | Done        |
+Documentation | [MISSING_FUNC][]        | Done        |
+Documentation | [FEATURE_FROZEN][]      | Done        | Based on currently known feature set
+RTL           | [FEATURE_COMPLETE][]    | Done        | Based on currently known feature set
```
Comment (imphil) [inline hw/ip/keymgr/doc/checklist.md:48]: Everything in the checklist refers to the "currently specified feature set", so you can omit that part. When the spec changes, just increase the version number and go down from D2 to D1 to indicate that the RTL now doesn't fully implement the spec any more.

Comment (tjaychen) [reply hw/ip/keymgr/doc/checklist.md:48]: thanks @imphil i'll do that.

File: hw/ip/keymgr/data/keymgr.prj.hjson (hunk 1 above)
Comment (tjaychen) [reply hw/ip/keymgr/data/keymgr.prj.hjson:15]: thanks @imphil i'll do that.

**Comment by tjaychen** [comment]
> LGTM with 2 questions
> 
> 1. the sideload interface to aes and hmac hasn't been finalized, right?
> 2. can you fix this keymgr related issue #4899?

o yeah thanks for the reminder, let me have another look.

**Comment by tjaychen** [comment]
i'm closing this PR down now since keymgr needs to go through a round of feature enhancements for sideload and DICE compliance.

**Pull request closed without merging**